src/couch_mrview/src/couch_mrview_debug.erl

```diff
@@ -48,3 +280,152 @@ view_signature(DbName, DDocName) ->
+                        {db_name, DbName},
+                        {idx_name, IdxState#mrst.idx_name},
+                        {view_file_path, couch_mrview_util:index_file(DbName, Sig)}
+                    ],
+            {ok, maps:from_list(State)};
+        _ ->
+            {error, not_mrview_index}
+    end.
+
+view_state(PidOrIdxState) ->
+    view_state(PidOrIdxState, undefined).
+
+view_state(Pid, ViewName) when is_pid(Pid) ->
+    {ok, IdxState} = couch_index:get_state(Pid, 0),
+    view_state(IdxState, ViewName);
+view_state(IdxState, ViewName) ->
+    case IdxState of
+        #mrst{} ->
+            MrViews = lists:foldl(
+                fun(MrView, Acc) ->
+                    {Name, ReduceFuns} =
+                        case MrView#mrview.reduce_funs of
+                            [] ->
+                                {hd(MrView#mrview.map_names), []};
+                            _ ->
+                                % reduce_funs contains tuples of {Name, ReduceFuns}
+                                hd(MrView#mrview.reduce_funs)
+                        end,
+                    View = #{
+                        id_num => MrView#mrview.id_num,
+                        update_seq => MrView#mrview.update_seq,
+                        purge_seq => MrView#mrview.purge_seq,
+                        reduce_funs => ReduceFuns,
+                        def => MrView#mrview.def,
+                        btree_size => couch_btree:size(MrView#mrview.btree),
+                        options => MrView#mrview.options
+                    },
+                    maps:put(Name, View, Acc)
+                end,
+                #{},
+                IdxState#mrst.views
+            ),
+            case ViewName of
+                undefined ->
+                    {ok, MrViews};
+                _ ->
+                    case maps:get(ViewName, MrViews) of
+                        {badkey, Key} ->
+                            io:format("No view named ~p was found.", [Key]),
+                            {error, {badkey, Key}};
+                        Value ->
+                            {ok, #{ViewName => Value}}
+                    end
+            end;
+        _ ->
+            {error, not_mrview_index}
+    end.
+
+index_view_state(Pid) when is_pid(Pid) ->
+    index_view_state(Pid, undefined).
+
+index_view_state(Pid, ViewName) when is_pid(Pid) ->
+    {ok, IdxState} = index_state(Pid),
+    {ok, ViewState} = view_state(Pid, ViewName),
+    {ok, maps:put(views, ViewState, IdxState)}.
+
+index_report(Pid) when is_pid(Pid) ->
+    case index_state(Pid) of
+        {ok, IdxState} ->
+            Sig = maps:get(signature, IdxState),
+            IdxState2 = maps:put(signature, couch_util:to_hex(Sig), IdxState),
+            % Convert collator versions to strings to print pretty
+            IdxState3 = convert_collator_versions_to_strings(IdxState2),
+            couch_debug:print_report_with_info_width(maps:to_list(IdxState3), 21);
```

Comment:
I've notices that `view_file_path` is quite long and truncated in the example of the output. Let's shorten `view_file_path` field by removing the common prefix.

```erlang
    ...
    IdxState4 = maps:update_with(view_file_path, fun format_view_path/1, IdxState3).
    ...
end.

format_view_path(ViewFilePath) ->
    BaseDir = config:get("couchdb", "view_index_dir"),
    lists:flatten(string:replace(ViewFilePath, BaseDir ++ "/", "")).
```